### PR TITLE
fix(authz): ensures extauthz provider is removed from control plane during cleanup

### PR DESCRIPTION
**Pull request by bartoszmajsak**
### Renames migration folder
The reason for this is to have a simple naming convention instead of suggesting storing migration patches in dedicated folders named after tickets.

Additionally, the feature explicitly orders files instead of assuming that the underlying fsys implementation fulfills such a contract.

### Ports #605 test for extension provider

This test ensures the addition of an extension provider for external authorization and that it is removed from the control plane properly using a custom cleanup function.

We have missed it in the original work.

### Fix: aligns provider name between template and cleanup logic

This is short term fix for existing codebase. Long term (which is actively worked on) we need to improve the way of how we are storing config information to limit cases where we rely on pre/suffixes. Cases like this should be kept as it's own thing instead, as it represents the concept in the infrastructure/authz setup.

### CODE REVIEW

**Review by zdtsw — APPROVED** [review]

File: tests/integration/features/servicemesh_feature_test.go [hunk 1]
```diff
@@ -43,136 +45,227 @@ var _ = Describe("Service Mesh feature", func() {
...
-		When("operator is installed", func() {
-			var smcpCrdObj *apiextensionsv1.CustomResourceDefinition
+			When("operator is installed", func() {
+				var smcpCrdObj *apiextensionsv1.CustomResourceDefinition
 
-			BeforeEach(func() {
-				err := fixtures.CreateSubscription(fixtures.OssmSubscription, "openshift-operators", envTestClient)
-				Expect(err).ToNot(HaveOccurred())
+				BeforeEach(func() {
+					err := fixtures.CreateSubscription(fixtures.OssmSubscription, "openshift-operators", envTestClient)
```
Comment (bartoszmajsak) [inline tests/integration/features/servicemesh_feature_test.go:80]: As tests run in sequence there is no need now to remove subscription, but I will clean this up in the follow-up PR.

**Review by israel-hdez — APPROVED** [review]
So we have the cleanup fixed and we have some additional units to make sure all is working and wired correctly.

/lgtm

**Pull request merged**